web/client/components/geostory/builder/SectionsPreview.jsx

```diff
@@ -6,104 +6,274 @@
+import { connect } from "react-redux";
+import { createSelector } from 'reselect';
+import uuid from "uuid";
+
+import TitleEditable from './TitleEditable';
 
+import { resourcesSelector } from '../../../selectors/geostory';
+import Toolbar from '../../misc/toolbar/Toolbar';
 import SideGrid from '../../misc/cardgrids/SideGrid';
 import SideCard from '../../misc/cardgrids/SideCard';
+import draggableContainer from '../../misc/enhancers/draggableContainer';
+import draggableComponent from '../../misc/enhancers/draggableComponent';
+
+const DraggableSideGrid = draggableContainer(SideGrid);
+const DraggableSideCard = draggableComponent((props) => <SideCard {...props} dragSymbol={<Glyphicon glyph="menu-hamburger"/>} />);
+
+/**
+ * avoid triggering other events like card selection
+ * @param {string} id id of the element to scroll to
+ * @param {function} scrollTo function used to scroll to the element
+ */
+const scrollToHandler = (id, scrollTo) => (evt) => {
+    evt.stopPropagation();
+    scrollTo(id);
+};
+
 
 /**
  * Renders icon for the Section or content provided
- * @param {object} content content or section of the type of the icon you want to render
+ * @prop {object} props passed to this icon
+ * @prop {string} props.type type of the icon to render
+ * @prop {string} props.src src of the image
+ * @prop {string} props.thumbnail url for the thumbnail of the map
  */
-const Icon = ({ type } = {}) => {
+const Icon = ({ type, src, thumbnail } = {}) => {
     const glyphs = {
-        text: 'sheet'
+        text: 'sheet',
+        image: 'picture',
+        title: 'story-title-section',
+        paragraph: 'story-paragraph-section',
+        immersive: 'story-immersive-section',
+        media: 'story-media-section',
+        map: '1-map',
+        columnleft: 'align-left',
+        columnright: 'align-right',
+        columncenter: 'align-center'
     };
-    return <Glyphicon glyph={glyphs[type] || 'question-sign'} />;
+    const imgSrc = src || thumbnail;
+    return imgSrc ? <img src={imgSrc}/> : <Glyphicon glyph={glyphs[type] || 'picture'} />;
 };
+const ConnectedIcon = compose(
+    connect(createSelector(resourcesSelector, (resources) => ({resources}))),
+    withProps(
+        ({ resources, resourceId: id }) => {
+            const resource = find(resources, { id }) || {};
+            return resource.data;
+        }
+    )
+)(Icon);
 
-/**
- * Preview view for a geostory-content
- * @param {object} [content={}] content that have to render preview
- */
-const Preview = ({ width } = {}) => {
-    const res = 1080 / 1920;
-    const height = width * res;
-    return (
-        <div className="ms-section-preview" style={{ width, height }}>
-            <div className="ms-section-preview-icon">
-                <Glyphicon glyph="1-map" style={{ fontSize: height / 2 }} />
-            </div>
+const previewContents = {
+    title: () => null,
+    paragraph: ({ id, contents, isCollapsed, scrollTo, onSort, sectionId, onUpdate }) => (
```

Comment:
Is the div container really needed? can you instead use less to make the draggablesidegrid position relative?

Reply:
this part will change soon, but i can investigate your suggestion